addons/spreadsheet_dashboard/tests/test_dashboard_data.py

```diff
@@ -12,3 +15,32 @@ def test_validate_dashboard_data(self):
         for dashboard in dashboards:
             with self.subTest(dashboard.name):
                 self.validate_spreadsheet_data(dashboard.raw, dashboard.name)
+
+    @mute_logger('odoo.sql_db')
+    def test_onchange_data_valid_json(self):
+        """Test onchange method for valid JSON data"""
+        group = self.env["spreadsheet.dashboard.group"].create(
+            {"name": "a group"}
+        )
+        spreadsheet = self.env["spreadsheet.dashboard"].create({
+            'name': 'Test Model',
+            'data': base64.b64encode(json.dumps({'test': 'data'}).encode('utf-8')),
+            'dashboard_group_id': group.id,
+        })
+        spreadsheet._onchange_data_()
```

Comment:
you can use the `Form` testing utility to test onchanges :wink: 
https://github.com/odoo/odoo/blob/16.0/odoo/tests/common.py#L1849